### ISSUE TITLE
sync waves are not honoured between app of apps' child applications after initial sync

### ISSUE THREAD

**Issue by KipsasJaujoj** [argoproj/argo-cd#8778]
* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

While using the app of apps pattern with child waves added, once child applications are deployed (initial sync is completed), changes to them are synchronized without following sync waves order.

**To Reproduce**

Enable custom health assessment to application CRD:
https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/health.md
(I've also tried the one provided on https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/1.7-1.8/#health-assessment-of-argoprojioapplication-crd-has-been-removed)

App of apps repository: https://github.com/KipsasJaujoj/app-of-apps
Uses manifests in argocd-example-apps. I forked the repository so I can update image tags and observe changes:
https://github.com/KipsasJaujoj/argocd-example-apps

Deploy app of apps and observe initial deployment:
`guestbook-one` sync is in progress and only once it is healthy, `guestbook-two` is in progress and eventually deployed.

Update image tags (or do any other changes) to child application manifests and push changes to repository.

Out of sync status is not picked in app of apps application view. Synchronization happens asychronously, without any order defined by sync waves.

**Expected behavior**

Once changes are done to child application manifests, their objects in app of apps view should show out of sync status. Synchronization order should be according to sync waves.

**Pictures**

Pictures below shows out of sync status in child application view but healthy and synced status in app of apps view.
![image (1)](https://user-images.githubusercontent.com/8336391/158256534-456bd845-13ae-476f-89ca-679e1a0202a2.png)
![image](https://user-images.githubusercontent.com/8336391/158256536-9ecc3cf0-97a4-44a5-b0e7-9786529d2fad.png)

**Version**

```shell
(default) argocd version
argocd: v2.1.1+aab9542.dirty
  BuildDate: 2021-08-31T16:14:11Z
  GitCommit: aab9542f8b3354f0940945c4295b67622f0af296
  GitTreeState: dirty
  GoVersion: go1.17
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v2.2.5+8f981cc
  BuildDate: 2022-02-05T01:33:25Z
  GitCommit: 8f981ccfcf942a9eb00bc466649f8499ba0455f5
  GitTreeState: clean
  GoVersion: go1.16.11
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v4.2.0 2021-06-30T22:49:26Z
  Helm Version: v3.7.1+g1d11fcb
  Kubectl Version: v0.22.2
  Jsonnet Version: v0.17.0
```

**Comment by rishabh625** [comment]
Apps of apps pattern can be used to deploy one application using another and syncwaves can be used to sync resources created by application in particular order, since in apps of apps pattern once a application(app1) creates another application(app2) app2 is treated not related to app1

**Comment by rishabh625** [comment]
@leoluz : this to be considered for feature development?

**Comment by leoluz** [comment]
I don't think this is a bug but how argocd was designed to work. In the Apps-of-Apps pattern the only resource managed by the parent app are the child Applications themselves. Image tags are not part of the Application resource and so are not reported in diffs. ArgoCD will always compare the desired state with the live state and in the apps-of-apps case, only the Application resource changes are validated.

Im closing this ticket for now. Feel free to reopen if you think this is a real issue.

**Comment by adam-harries** [comment]
> @leoluz Hi, I'm trying to solve the same issue. I'd like to use SyncWaves for CD of apps in waves. When we introduce na new version I'd like apps to update gradually, in waves. This only happes if at the creation. After, when the child apps are out of sync this behaviour doesn't propagate to parent app. Like you said `only the Application resource changes are validated.` And I'd like them to sync gradually, in waves also when updating their config (helmchart values files).
> 
> Did you find any viable solution? I'm starting to feel depresed and deparate as I can't figure this out :D

@mknapcok I'm seeing the same issue. Did you ever find a workaround for this?
